pkg/notify/notifier/sms/aws.go

```diff
@@ -0,0 +1,92 @@
+package sms
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/sns"
+	"github.com/kubesphere/notification-manager/pkg/apis/v2beta2"
+	"github.com/kubesphere/notification-manager/pkg/controller"
+	"github.com/kubesphere/notification-manager/pkg/utils"
+)
+
+const (
+	awsMaxPhoneNums = 1000
+)
+
+type AWSNotifier struct {
+	notifierCtl     *controller.Controller
+	AccessKeyId     *v2beta2.Credential
+	SecretAccessKey *v2beta2.Credential
+	PhoneNums       string
+	Region          string
+}
+
+func NewAWSProvider(c *controller.Controller, providers *v2beta2.Providers, phoneNumbers []string) Provider {
+	phoneNums := handleAWSPhoneNums(phoneNumbers)
+	return &AWSNotifier{
+		notifierCtl:     c,
+		AccessKeyId:     providers.AWS.AccessKeyId,
+		SecretAccessKey: providers.AWS.SecretAccessKey,
+		PhoneNums:       phoneNums,
+		Region:          providers.AWS.Region,
+	}
+}
+
+type SNSPublishAPI interface {
+	Publish(ctx context.Context,
+		params *sns.PublishInput,
+		optFns ...func(*sns.Options)) (*sns.PublishOutput, error)
+}
+
+func PublishMessage(c context.Context, api SNSPublishAPI, input *sns.PublishInput) (*sns.PublishOutput, error) {
+	return api.Publish(c, input)
+}
+
+func (a *AWSNotifier) MakeRequest(_ context.Context, messages string) error {
+	accessKeyId, err := a.notifierCtl.GetCredential(a.AccessKeyId)
+	if err != nil {
+		return utils.Errorf("[AWS  SendSms] cannot get accessKeyId: %s", err.Error())
+	}
+	secretAccessKey, err := a.notifierCtl.GetCredential(a.SecretAccessKey)
+	if err != nil {
+		return utils.Errorf("[AWS  SendSms] cannot get secretAccessKey: %s", err.Error())
+	}
+
+	err = os.Setenv("AWS_ACCESS_KEY_ID", accessKeyId)
```

Comment:
If there are two configs with different accessKeyId, this environment variable will be overwritten, is this as expected?

Reply:
If there is a receiver `a` with config `a` like this.

```
spec:
  sms:
    aws:
      accessKeyId: 
        value: accessKeyId1
      secretAccessKey:
        value: secretAccessKey1
```

and a receiver `b` with config `b` like this.

```
spec:
  sms:
    aws:
      accessKeyId: 
        value: accessKeyId2
      secretAccessKey:
        value: secretAccessKey2
```

If these two receivers send notifications at the same time, the `AWS_ACCESS_KEY_ID` will be set twice? What is the value of `AWS_ACCESS_KEY_ID` when sending the short message, `accessKeyId1` or `accessKeyId2`?

Reply:
Good question, I'll try to resolve it.